File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.cpp
#include <vector>

using namespace std;
using ll = long long;

vector<int> solution(int n, ll left, ll right) {
  vector<int> answer;

  for (ll i = left; i < right + 1; i++) answer.push_back(max(i / n, i % n) + 1);
  return answer;
}

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.java
class Solution {

    public int[] solution(int n, long left, long right) {
        int size = (int) (right - left + 1);
        int[] answer = new int[size];

        for (int i = 0; i < size; i++) {
            long index = left + i;
            answer[i] = (int) (Math.max(index / n, index % n) + 1);
        }

        return answer;
    }
}